Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc

```diff
@@ -195,10 +196,11 @@ AlCaIsoTracksProducer::AlCaIsoTracksProducer(const edm::ParameterSet& iConfig) :
   trigKount = trigPass = dummy;
 
   //create also IsolatedPixelTrackCandidateCollection which contains isolation info and reference to primary track
-  produces<reco::HcalIsolatedTrackCandidateCollection>("HcalIsolatedTrackCollection");
+  static const std::string labelIsoTk = "HcalIsolatedTrackCollection";
```

Comment:
@bsunanda almost there.
There is still the magic string `HcalIsolatedTrackCollection` in the `Event::put` call in the `produce` method. See
https://github.com/bsunanda/cmssw/blob/Run2-alca14/Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc#L340
The `std::string labelIsoTk` should be not a `const` in the c'tor body, but a `static const` data member of the producer.
Something like:

````
 // producer interface
 //...
  edm::EDGetTokenT<EcalRecHitCollection>                  tok_EE_;
  edm::EDGetTokenT<HBHERecHitCollection>                  tok_hbhe_;
  static const std::string labelIsoTk;

// producer body
const std::string AlCaIsoTracksProducer::labelIsoTk = "HcalIsolatedTrackCollection";

AlCaIsoTracksProducer::AlCaIsoTracksProducer(const edm::ParameterSet& iConfig)
// in the c'tor body, remove the declaration in this line
//...
  edm::LogInfo("HcalIsoTrack") << " Expected to produce the collections:\n"
			       << "reco::HcalIsolatedTrackCandidateCollection "
			       << " with label " << labelIsoTk << "\n"
			       << "reco::VertexCollection with label " << labelRecVtx_.label() << "\n"
			       << "EcalRecHitCollection with label EcalRecHitsEB\n"
			       << "EcalRecHitCollection with label EcalRecHitsEE\n"
			       << "HBHERecHitCollection with label " << labelHBHE_.label();

void AlCaIsoTracksProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
// in the produce method, use the string for designating the collection to be put in the event
  iEvent.put(outputHcalIsoTrackColl, labelIsoTk);
````